core-metastore/src/main/java/org/zalando/nakadi/repository/db/EventTypeRepository.java

```diff
@@ -103,6 +105,15 @@ public EventType mapRow(final ResultSet rs, final int rowNum) throws SQLExceptio
                 new EventTypeMapper());
     }
 
+    public List<EventType> listEventTypesWithRowLock(final Set<String> eventTypes) {
+        final String whereClause = "WHERE zn_data.event_type.et_name in ( "
+                + String.join(",", Collections.nCopies(eventTypes.size(), "?") ) +")";
+        return jdbcTemplate.query(
+                "SELECT et_event_type_object FROM zn_data.event_type " + whereClause + " FOR KEY SHARE",
```

Comment:
Looks like a lot of string operations.  Could we make use `String.format()` instead of most concatenations here?

Reply:
done